### PR TITLE
fix substr crash

### PR DESCRIPTION
**Pull request by nevermore3**
refer to https://github.com/vesoft-inc/nebula-graph/issues/878

### CODE REVIEW

File: src/common/function/FunctionManager.cpp [hunk 1]
```diff
@@ -1313,32 +1313,26 @@ FunctionManager::FunctionManager() {
         attr.isPure_ = true;
         attr.body_ = [](const auto &args) -> Value {
             auto argSize = args.size();
-            if (argSize < 2 || argSize >3) {
+            if (argSize < 2 || argSize > 3) {
```
Comment (Shylock-Hg) [inline src/common/function/FunctionManager.cpp:1316]: The find method will check the count of arguments.

File: src/common/function/FunctionManager.cpp [hunk 2]
```diff
@@ -1313,32 +1313,26 @@ FunctionManager::FunctionManager() {
...
                 LOG(ERROR) <<  "Unexpected arguments count " << args.size();
                 return Value::kNullBadData;
             }
+            if (args[0].isNull()) {
```
Comment (laura-ding) [inline src/common/function/FunctionManager.cpp:1320]: `empty` is the same. but the type Signature has made sure the input args[0] must string.

File: src/common/function/FunctionManager.cpp (hunk 1 above)
Comment (nevermore3) [reply src/common/function/FunctionManager.cpp:1316]: fix

File: src/common/function/FunctionManager.cpp (hunk 2 above)
Comment (nevermore3) [reply src/common/function/FunctionManager.cpp:1320]: type Signature only use in deduce expr's type , When it encounters null, it returns directly in validator
So in the execution stage, we need to re-judge

user cannot enter EMPTY .so we not process EMPTY

Comment (laura-ding) [reply src/common/function/FunctionManager.cpp:1316]: the query input can be empty, such as pipe

Comment (nevermore3) [reply src/common/function/FunctionManager.cpp:1316]: EMPTY will do later ,all function have this problem

Comment (laura-ding) [reply src/common/function/FunctionManager.cpp:1316]: Okay

File: src/common/function/test/FunctionManagerTest.cpp [hunk 3]
```diff
@@ -150,6 +150,11 @@ TEST_F(FunctionManagerTest, testNull) {
     TEST_FUNCTION(toUpper, args_["nullvalue"], Value::kNullValue);
     TEST_FUNCTION(split, std::vector<Value>({Value::kNullValue, ","}), Value::kNullValue);
     TEST_FUNCTION(split, std::vector<Value>({"123,22", Value::kNullValue}), Value::kNullValue);
+    TEST_FUNCTION(substr, std::vector<Value>({Value::kNullValue, 1, 2}), Value::kNullValue);
+    TEST_FUNCTION(substr, std::vector<Value>({"hello", Value::kNullValue, 2}), Value::kNullBadType);
+    TEST_FUNCTION(substr, std::vector<Value>({"hello", 2, Value::kNullValue}), Value::kNullBadType);
+    TEST_FUNCTION(substr, std::vector<Value>({"hello", -1, 10}), Value::kNullBadData);
+    TEST_FUNCTION(substr, std::vector<Value>({"hello", 1, -2}), Value::kNullBadData);
```
Comment (laura-ding) [inline src/common/function/test/FunctionManagerTest.cpp:157]: add test when `start` large than string's size

Comment (nevermore3) [reply src/common/function/test/FunctionManagerTest.cpp:157]: alread exist in line 244

**Review by laura-ding — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**